### ISSUE TITLE
jenkins_spec_check doesn't properly expand variables in Dockerfiles

### ISSUE THREAD

**Issue by alanhoyle** [BioContainers/containers#296]
I'm trying to contribute a Dockerfile to build samtools 1.7.0 #295.  The Dockerfile I created uses an ARG variable for the software_version, and builds successfully in a real Docker environment, but when I try to do a pull request, the automated build fails on your end.  The problem seems to be that the jenkins_spec_check does not properly expand the ARG variable in the file.  I think that using an ARG variable is superior to the current method for two reasons:

1.  It would allow adding a new version/Dockerfile dedicated to a specific version of software by changing only one line in the Dockerfile, instead of the more error-prone requirement that it be changed on two lines.
2.  It would allow an end-user to attempt to build their own version by adding a `--build-arg` parameter to their local build command (i.e. `docker build software_version=$VERSION_HERE -t [...]`).  

```
==> Dockerfile.with_ARG_line <==
################## BASE IMAGE ######################

FROM biocontainers/biocontainers:latest

################## METADATA ######################

ARG software_version=1.7.0
LABEL base_image="biocontainers:latest"
LABEL version="2"
LABEL software="Samtools"
LABEL software.version="$software_version"
LABEL about.summary="Tools for manipulating next-generation sequencing data"
LABEL about.home="https://github.com/samtools/samtools"
LABEL about.documentation="https://github.com/samtools/samtools"
LABEL about.license_file="https://github.com/samtools/samtools"
LABEL about.license_file="https://github.com/samtools/samtools/blob/develop/LICENSE"
LABEL about.license="SPDX:MIT"
LABEL about.tags="Genomics"
LABEL extra.identifiers.biotools="SAM"

################## MAINTAINER ######################
MAINTAINER Saulo Alves Aflitos <sauloal@gmail.com>

RUN conda install ncurses samtools=$software_version

WORKDIR /data/

CMD ["samtools"]
```
vs:
```
=> Dockerfile <==
################## BASE IMAGE ######################

FROM biocontainers/biocontainers:latest

################## METADATA ######################

LABEL base_image="biocontainers:latest"
LABEL version="2"
LABEL software="Samtools"
LABEL software.version="1.7.0"
LABEL about.summary="Tools for manipulating next-generation sequencing data"
LABEL about.home="https://github.com/samtools/samtools"
LABEL about.documentation="https://github.com/samtools/samtools"
LABEL about.license_file="https://github.com/samtools/samtools"
LABEL about.license_file="https://github.com/samtools/samtools/blob/develop/LICENSE"
LABEL about.license="SPDX:MIT"
LABEL about.tags="Genomics"
LABEL extra.identifiers.biotools="SAM"

################## MAINTAINER ######################
MAINTAINER Saulo Alves Aflitos <sauloal@gmail.com>

RUN conda install ncurses samtools=1.7.0

WORKDIR /data/

CMD ["samtools"]
```

**Comment by osallou** [comment]
I'll see if there is a way to manage this.
Docker parse.library does not handle arg templating.

**Comment by osallou** [comment]
fix in ci check scripts: BioContainers/ci/commit/0975fc09f337bc9b19f07a60f730995a169fdef5
